app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -256,7 +284,31 @@ class ShowAlternativeRoutesActivity : AppCompatActivity() {
                         .build()
                 )
                 .build()
-        )
+        ).attach(this)
```

Comment:
We should consider that when you quickly go from one example to another, `onCreate` of the second example may run before `onDestroy` of the first one. This means that `MapboxNavigation` instance will be reused between these two examples, which shouldn't happen.

Reply:
Yeah this is a little complicated. Activity examples are demonstrating "stand-alone" examples. Examples where Activity switching is happening, there are ways to utilize `MapboxNavigationApp` to make the experience seamless, because the `MapboxNavigation` instance is not recreated.

Also note that, the newest Activity that calls `MapboxNavigationApp.setup(` in `onCreate` will device what the `NavigationOptions` are.

Reply:
Not sure I understand. I see that the second `MapboxNavigationApp.setup` call is ignored, unless you call `MapboxNavigationApp.disable` before that. Clearly we don't want the second example to reuse navigation options from the first one. Maybe we should call `MapboxNavigationApp.disable` before calling `MapboxNavigationApp.setup`.

Reply:
Going directly from one example to another will be complicated.

But yeah, I agree with you. A simple solution would be to add `disable` to the `MainActivity`. I was thinking about an example where we are going directly between examples.

Reply:
But also, when all attached `LifecycleOwner`s are destroyed, the MapboxNavigation instance will be destroyed too. So a `disable` call may not be needed 🤔

Reply:
`disable` is needed, because the second `setup` call is ignored. It doesn't matter if there are any `LifecycleOwner`s created at the moment. You will even see this log:
https://github.com/mapbox/mapbox-navigation-android/blob/0c1d2c74c360b52afcbbfa1937ad52f373a61bef/libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/MapboxNavigationAppDelegate.kt#L27-L35.

Reply:
Ok seems best to add a `disable` call in `MainActivity`.